spec/upstream_spec.lua

```diff
@@ -66,29 +66,44 @@ describe('Upstream', function()
         end)
     end)
 
-    describe(':set_request_host', function()
-        it('sets request Host header to the URI host', function()
-            stub(ngx.req, 'set_header')
+    local function stub_ngx_request()
+        ngx.var = { }
+
+        stub(ngx, 'exec')
+        stub(ngx.req, 'set_header')
+        stub(ngx.req, 'set_uri', function(uri)
+            ngx.var.uri = uri
+        end)
+        stub(ngx.req, 'set_uri_args', function(args)
+            ngx.var.args = args
+            ngx.var.query_string = args
+        end)
+    end
+
+    describe(':rewrite_request', function()
```

Comment:
I think we should add tests for this part of the code too:
```lua
    if uri.query then
        ngx.req.set_uri_args(uri.query)
    end
```

Reply:
Done.